packages/agent-toolkit/src/shared/parameters.ts

```diff
@@ -9,123 +11,127 @@ const addressSchema = z.string().regex(/^0x[a-fA-F0-9]{40}$/);
  * Parameters for the getAccountInfo function
  * @param address - The address of the account to get account info for (optional)
  */
-export const getAccountInfoParameters = z.object({
-  address: addressSchema
-    .optional()
-    .describe("The address of the account (EVM hex string address)"),
-});
+export const getAccountInfoParameters = (_context: Context = {}) =>
+  z.object({
+    address: addressSchema
+      .optional()
+      .describe("The address of the account (EVM hex string address)"),
+  });
 
 /**
  * Parameters for the listBuckets function
  * @param address - The address of the account to list buckets for (optional)
  */
-export const listBucketsParameters = z.object({
-  address: addressSchema
-    .optional()
-    .describe(
-      "The address of the account to list buckets for (EVM hex string address)",
-    ),
-});
+export const listBucketsParameters = (_context: Context = {}) =>
+  z.object({
+    address: addressSchema
+      .optional()
+      .describe(
+        "The address of the account to list buckets for (EVM hex string address)",
+      ),
+  });
 
 /**
  * Parameters for the getCreditInfo function
  * @param address - The address of the account to get credit info for (optional)
  */
-export const getCreditInfoParameters = z.object({
-  address: addressSchema
-    .optional()
-    .describe("The address of the account (EVM hex string address)"),
-});
+export const getCreditInfoParameters = (_context: Context = {}) =>
+  z.object({
+    address: addressSchema
+      .optional()
+      .describe("The address of the account (EVM hex string address)"),
+  });
 
 /**
  * Parameters for the buyCredit function
  * @param amount - The amount of credit to buy
  * @param to - The address of the account to buy credit for (optional)
  */
-export const buyCreditParameters = z.object({
-  amount: z.string().min(1).describe("The amount of credit to buy"),
-  to: addressSchema
-    .optional()
-    .describe(
-      "The address of the account to buy credit for (EVM hex string address)",
-    ),
-});
+export const buyCreditParameters = (_context: Context = {}) =>
+  z.object({
+    amount: z.string().min(1).describe("The amount of credit to buy"),
+    to: addressSchema
+      .optional()
+      .describe(
+        "The address of the account to buy credit for (EVM hex string address)",
+      ),
+  });
 
 /**
  * Parameters for the createBucket function
  * @param bucketAlias - The alias of the bucket to create
  * @param metadata - The metadata to store with the bucket (optional)
  */
-export const createBucketParameters = z.object({
-  bucketAlias: z.string().min(1).describe("The alias of the bucket to create"),
-  metadata: z
-    .record(
-      z.string().min(1).describe("The key of the metadata"),
-      z.string().min(1).describe("The value of the metadata"),
-    )
-    .optional()
-    .describe("The metadata to store with the bucket"),
-});
+export const createBucketParameters = (_context: Context = {}) =>
```

Comment:
allow `Context` to get passed...tbd on if/what we use, but "global" bucket aliases are a good example.